staging/src/k8s.io/apiserver/pkg/endpoints/filters/tenantinfo.go

```diff
@@ -28,44 +28,48 @@ import (
 // WithTenantInfo set the tenant in the requestInfo for short-path requests based on the user info from the authentication result.
 func WithTenantInfo(handler http.Handler) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
-		ctx := req.Context()
 
-		requestor, exists := request.UserFrom(ctx)
-		if !exists {
-			responsewriters.InternalError(w, req, errors.New("The user info is missing."))
+		newReq, err := NormalizeTenant(req)
+		if err != nil {
+			responsewriters.InternalError(w, req, err)
 			return
 		}
 
-		tenantInRequestor := requestor.GetTenant()
-		if tenantInRequestor == metav1.TenantNone {
-			// temporary workaround
-			// tracking issue: https://github.com/futurewei-cloud/arktos/issues/102
-			tenantInRequestor = metav1.TenantSystem
-			//responsewriters.InternalError(w, req, errors.New(fmt.Sprintf("The tenant in the user info of %s is empty. ", requestor.GetName())))
-			//return
-		}
+		handler.ServeHTTP(w, newReq)
+	})
+}
 
-		requestInfo, exists := request.RequestInfoFrom(ctx)
-		if !exists {
-			responsewriters.InternalError(w, req, errors.New("The request info is missing."))
-			return
-		}
+// NormalizeObjectTenant sets the tenant in request info based on the user tenant.
+func NormalizeTenant(req *http.Request) (*http.Request, error) {
```

Comment:
the name "Normalize" is quite confusing for me. Can we change this to something more straightforward like SetTenantInRequest?

Reply:
+1.  "Normalize" has special meanings.

Reply:
updated the func name to SetShortPathRequestTenant.